tests/unit/valgrind.suppressions

```diff
@@ -10,3 +10,51 @@
    fun:pthread_create@@GLIBC_2.2.5
    fun:main
 }
+
+# --errors-for-leak-kinds=all reports pedantic `reachable` leaks so we
+# suppress some trivial ones here
+{
+   ignore_fprint_allocations
+   Memcheck:Leak
+   match-leak-kinds: reachable
+   ...
+   fun:fprintf
+   fun:main
+}
+{
+   ignore_setenv_allocations
+   Memcheck:Leak
+   match-leak-kinds: reachable
+   ...
+   fun:setenv
+   fun:main
+}
+
+
+# TODO: fix the pedantic leak errors from s2n_fork_generation_number_test
```

Comment:
Does this need an issue? (not sure my old https://github.com/aws/s2n-tls/issues/3334 applies)

Reply:
It wasnt flaky for me. The cleanup just wasnt running and I wasn't able to figure out a solution. I can add a new issue.

Reply:
https://github.com/aws/s2n-tls/issues/3758